tests/testthat/test-sector_profile_at_product_level.R

```diff
@@ -218,3 +218,13 @@ test_that("`*rowid` columns are passed through inputs with duplicates", {
   expect_true(hasName(out, "companies_rowid"))
   expect_true(hasName(out, "scenarios_rowid"))
 })
+
+test_that("yields non-missing `clustered` when `risk_category` is `NA` (#587)", {
+  companies <- example_companies()
+  scenarios <- example_scenarios(!!aka("co2reduce") := NA)
+
+  out <- sector_profile_at_product_level(companies, scenarios)
+
+  expect_true(is.na(out$risk_category))
+  expect_false(is.na(out$clustered))
+})
```

Comment:
@AnneSchoenauer just FYI,

You showed interest in learning how to read tests, and I said you can think of them as a reprex. Now you're familiar with reprex, and also with this particular PR. So I think this is a good example for you to attempt reading the test that I highlight here. 

The same code as a reprex is this:

``` r
devtools::load_all()
#> ℹ Loading tiltIndicator

companies <- example_companies()
companies
#> # A tibble: 1 × 8
#>   company_id clustered activity_uuid_product_uuid sector subsector tilt_sector
#>   <chr>      <chr>     <chr>                      <chr>  <chr>     <chr>      
#> 1 a          a         a                          total  energy    a          
#> # ℹ 2 more variables: tilt_subsector <chr>, type <chr>

scenarios <- example_scenarios(!!aka("co2reduce") := NA)
scenarios
#> # A tibble: 1 × 6
#>   reductions sector subsector year  type  scenario
#>   <lgl>      <chr>  <chr>     <chr> <chr> <chr>   
#> 1 NA         total  energy    2050  ipr   a

out <- sector_profile_at_product_level(companies, scenarios)
out
#> # A tibble: 1 × 11
#>   companies_id grouped_by risk_category profile_ranking clustered
#>   <chr>        <chr>      <chr>         <lgl>           <chr>    
#> 1 a            <NA>       <NA>          NA              a        
#> # ℹ 6 more variables: activity_uuid_product_uuid <chr>, tilt_sector <chr>,
#> #   scenario <chr>, year <chr>, type <chr>, tilt_subsector <chr>

is.na(out$risk_category)
#> [1] TRUE

!is.na(out$clustered)
#> [1] TRUE
```

<sup>Created on 2023-11-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Thanks @maurolepore. That makes a lot of sense. And thanks also for the explanations! 

I have one question - so the test is now the at the end the test is.na(out$risk_category) TRUE and is.na(out$risk_category) FALSE. If the second would be TRUE then there would be  a mistake right? What is then happening? Is then there an error and the code stops? 

I have one quick question (which I guess should have seen before) but your scenarios tibble includes NA for sector, subsector, year, type and scenario. Is this example data? Because this shouldn't be the case (maybe a new requirement that we can test for?). I.e. if there is a scenario then there should always be reductions values. However, here it is NA.